src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/hotspot/HotSpotConstantPool.java

```diff
@@ -549,21 +549,27 @@ static class CachedBSMArgs extends AbstractList<JavaConstant> {
         /**
          * Lazily resolves and caches the argument at the given index and returns it. The method
          * {@link CompilerToVM#bootstrapArgumentIndexAt} is used to obtain the constant pool
-         * index of the entry and the method {@link CompilerToVM#lookupConstantInPool} is used
-         * to resolve it. If the resolution failed, the index is returned as a
+         * index of the entry and the method {@link ConstantPool#lookupConstant} is used to
+         * resolve it. If the resolution failed, the index is returned as a
          * {@link PrimitiveConstant}.
          *
          * @param index index of the element to return
-         * @return A {@link PrimitiveConstant} representing a {@code CONSTANT_Dynamic_info}
-         *         entry or a {@link JavaConstant} representing the static argument requested
+         * @return A {@link PrimitiveConstant} representing an unresolved constant pool entry
+         * or a {@link JavaConstant} representing the static argument requested
```

Comment:
`or a {@link JavaConstant}`
a PrimitiveConstant is a JavaConstant isn't it?

Reply:
> a PrimitiveConstant is a JavaConstant isn't it?
Yes, it is, but since the primitives are returned in their boxed form, a PrimitiveConstant can only be a constant pool index, so I think it is better to specify the difference.

Reply:
My point is that a PrimitiveConstant is a subtype of JavaConstant so the "or" does not make sense. It's like saying "an int or a number".